scripts/engineering/derive_phase44d_carry_aware_family_constant_surface.py

```
@@ -0,0 +1,580 @@
+    return 2 * spec.rolling_kv_pairs * spec.pair_width + 7 * spec.pair_width + 43
+
+
+def phase43_projection_columns(spec: FamilySpec) -> int:
+    return PHASE43_PROJECTION_PREFIX_WIDTH + spec.pair_width + 6 * PHASE43_PROJECTION_HASH_LIMBS
+
+
+def read_rows(path: Path) -> list[dict[str, str]]:
+    with path.open("r", encoding="utf-8") as handle:
+        rows = list(csv.DictReader(handle, delimiter="\t"))
+    if not rows:
+        raise SystemExit(f"no rows found in {path}")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)